### PR TITLE
Generate Appliance ResourceConnector dotnet SDK

### PR DESCRIPTION
**Pull request by aditimalladi**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/resourceconnector/Azure.ResourceManager.ResourceConnector/CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
+# Release History
+
+## 1.0.0-beta.1 (Unreleased)
```
Comment (ArthurMa1978) [inline sdk/resourceconnector/Azure.ResourceManager.ResourceConnector/CHANGELOG.md:3]: Specify the release date by replacing the `Unreleased`, the format is `yyyy-mm-dd`.

File: sdk/resourceconnector/Azure.ResourceManager.ResourceConnector/README.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
+# Microsoft Azure Azure.ResourceManager.ResourceConnector management client library for .NET
+
+**[Describe the service briefly first.]**
```
Comment (ArthurMa1978) [inline sdk/resourceconnector/Azure.ResourceManager.ResourceConnector/README.md:3]: Update this line for your service.

File: sdk/resourceconnector/Azure.ResourceManager.ResourceConnector/src/Properties/AssemblyInfo.cs [hunk 3]
```diff
@@ -0,0 +1,9 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Runtime.CompilerServices;
+
+[assembly: Azure.Core.AzureResourceProviderNamespace("Azure.ResourceManager.ResourceConnector")]
```
Comment (ArthurMa1978) [inline sdk/resourceconnector/Azure.ResourceManager.ResourceConnector/src/Properties/AssemblyInfo.cs:6]: The value here is `Microsoft.ResourceConnector`

File: sdk/resourceconnector/Azure.ResourceManager.ResourceConnector/src/autorest.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
+# Generated code configuration
+
+Run `dotnet build /t:GenerateCode` to generate code.
+
+``` yaml
+azure-arm: true
+csharp: true
+library-name: Azure.ResourceManager.ResourceConnector
+namespace: Azure.ResourceManager.Azure.ResourceManager.ResourceConnector
```
Comment (ArthurMa1978) [inline sdk/resourceconnector/Azure.ResourceManager.ResourceConnector/src/autorest.md:9]: the namespace here is `Azure.ResourceManager.ResourceConnector`

File: sdk/resourceconnector/ci.mgmt.yml [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+pr:
+  branches:
+    include:
+    - main
+    - feature/*
+    - hotfix/*
+    - release/*
+  paths:
+    include:
+    - sdk/azure.resourcemanager.resourceconnector/ci.mgmt.yml
```
Comment (ArthurMa1978) [inline sdk/resourceconnector/ci.mgmt.yml:15]: the folder is `sdk/resourceconnector/`

File: sdk/resourceconnector/ci.mgmt.yml [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+    - sdk/azure.resourcemanager.resourceconnector/Azure.ResourceManager.ResourceConnector/
+
+extends:
+  template: /eng/pipelines/templates/stages/archetype-sdk-client.yml
+  parameters:
+    ServiceDirectory: azure.resourcemanager.resourceconnector
```
Comment (ArthurMa1978) [inline sdk/resourceconnector/ci.mgmt.yml:33]: ServiceDirectory: resourcemanager

File: sdk/resourceconnector/ci.mgmt.yml [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+  branches:
+    include:
+    - main
+    - feature/*
+    - hotfix/*
+    - release/*
+  paths:
+    include:
+    - sdk/azure.resourcemanager.resourceconnector/ci.mgmt.yml
+    - sdk/azure.resourcemanager.resourceconnector/Azure.ResourceManager.ResourceConnector/
```
Comment (ArthurMa1978) [inline sdk/resourceconnector/ci.mgmt.yml:16]: - sdk/resourcemanager/Azure.ResourceManager.ResourceConnector/

File: sdk/resourceconnector/ci.mgmt.yml [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+
+
+  branches:
```
Comment (ArthurMa1978) [inline sdk/resourceconnector/ci.mgmt.yml:19]: remove ln 19 - 28 which is dup

File: sdk/resourceconnector/Azure.ResourceManager.ResourceConnector/tests/Azure.ResourceManager.ResourceConnectorManagementTestBase.cs [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+using Azure.Core;
+using Azure.Core.TestFramework;
+using Azure.ResourceManager.Resources;
+using Azure.ResourceManager.TestFramework;
+using NUnit.Framework;
+using System.Threading.Tasks;
+
+namespace Azure.ResourceManager.ResourceConnector.Tests
+{
+    public class Azure.ResourceManager.ResourceConnectorManagementTestBase : ManagementRecordedTestBase<Azure.ResourceManager.ResourceConnectorManagementTestEnvironment>
```
Comment (ArthurMa1978) [inline sdk/resourceconnector/Azure.ResourceManager.ResourceConnector/tests/Azure.ResourceManager.ResourceConnectorManagementTestBase.cs:13]: Please follow https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/resourcemanager/Azure.ResourceManager/docs/TestGuide.md to add tests for all resources.

File: sdk/resourceconnector/ci.mgmt.yml (hunk 6 above)
Comment (aditimalladi) [reply sdk/resourceconnector/ci.mgmt.yml:33]: Would this be the `resourcemanager` or the provider name here ? `resourceconnector`?

File: sdk/resourceconnector/Azure.ResourceManager.ResourceConnector/CHANGELOG.md [hunk 10]
```diff
@@ -0,0 +1,17 @@
+# Release History
+
+## 1.0.0 (2023-08-25)
```
Comment (ArthurMa1978) [inline sdk/resourceconnector/Azure.ResourceManager.ResourceConnector/CHANGELOG.md:3]: This is new package, so we need follow this [policy](https://azure.github.io/azure-sdk/policies_releases.html#beta-releases-and-stable-graduation) to release a beta first then stable after one month.

File: sdk/resourceconnector/ci.mgmt.yml (hunk 6 above)
Comment (ArthurMa1978) [reply sdk/resourceconnector/ci.mgmt.yml]: You are right, this is `resourceconnector`

File: sdk/resourceconnector/Azure.ResourceManager.ResourceConnector/CHANGELOG.md (hunk 10 above)
Comment (aditimalladi) [reply sdk/resourceconnector/Azure.ResourceManager.ResourceConnector/CHANGELOG.md]: Updated this accordingly

File: sdk/resourceconnector/Azure.ResourceManager.ResourceConnector/tests/Azure.ResourceManager.ResourceConnectorManagementTestBase.cs (hunk 9 above)
Comment (aditimalladi) [reply sdk/resourceconnector/Azure.ResourceManager.ResourceConnector/tests/Azure.ResourceManager.ResourceConnectorManagementTestBase.cs]: Added tests - working on getting tag updated on `assests.json`

File: sdk/resourceconnector/Azure.ResourceManager.ResourceConnector/src/Generated/Models/Distro.cs [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+
+#nullable disable
+
+using System;
+using System.ComponentModel;
+
+namespace Azure.ResourceManager.ResourceConnector.Models
+{
+    /// <summary> Represents a supported Fabric/Infra. (AKSEdge etc...). </summary>
+    public readonly partial struct Distro : IEquatable<Distro>
```
Comment (archerzz) [inline sdk/resourceconnector/Azure.ResourceManager.ResourceConnector/src/Generated/Models/Distro.cs:14]: Avoid using single word which is too generic. Consider appending RP prefix or renaming it.
- use `prepend-rp-prefix` to automatically append PR name. See example [here](https://github.com/Azure/azure-sdk-for-net/blob/176fd476588e8e61ebce747d66714cb39947cae5/sdk/azurestackhci/Azure.ResourceManager.Hci/src/autorest.md?plain=1#L52-L57).
- use `rename-mapping` to rename a model. See example [here](https://github.com/Azure/azure-sdk-for-net/blob/176fd476588e8e61ebce747d66714cb39947cae5/sdk/azurestackhci/Azure.ResourceManager.Hci/src/autorest.md?plain=1#L58-L76)

Comment (archerzz) [reply sdk/resourceconnector/Azure.ResourceManager.ResourceConnector/src/Generated/Models/Distro.cs:14]: That applies to other types like `SSHKey`

Comment (archerzz) [reply sdk/resourceconnector/Azure.ResourceManager.ResourceConnector/src/Generated/Models/Distro.cs:14]: > Not sure I understand the ask here? Is this to rename the Generated SDK model? That would not be possible right as this is derived from the Swagger. Is the ask here to update the `autorest.md` for naming convention? (Wondering if this should be left as to be uniform with other SDKs? )

@aditimalladi Yes. The comment is about naming conventions to follow [Azure .NET SDK guidelines](https://azure.github.io/azure-sdk/dotnet_introduction.html). Serialization logic will ensure the names in JSON will still be the ones defined in swagger.  You can take a look at the links I posted above.

File: sdk/resourceconnector/Azure.ResourceManager.ResourceConnector/tests/Azure.ResourceManager.ResourceConnectorManagementTestBase.cs (hunk 9 above)
Comment (aditimalladi) [reply sdk/resourceconnector/Azure.ResourceManager.ResourceConnector/tests/Azure.ResourceManager.ResourceConnectorManagementTestBase.cs]: Pushed the `assests.json` tag

**Review by ArthurMa1978 — APPROVED** [review]

**Pull request merged**